### PR TITLE
fix: Added multiselect type inconsistency in ```negotiate```  method

### PR DESCRIPTION
**Pull request by unniznd**
## What was wrong?
The multiselect type was inconsistent

## How was it fixed?
Set the return type to None and handled the places where the ```negotiate``` function is used.

Summary of approach.

### To-Do

- [ ] Clean up commit history
- [ ] Add or update documentation related to these changes
- [ ] Add entry to the [release notes](https://github.com/libp2p/py-libp2p/blob/main/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by pacrob** [comment]
Looks good, @unniznd. Please add a newsfragment and it should be good to go.

**Comment by unniznd** [comment]
@pacrob Added the newsfragment.

**Comment by unniznd** [comment]
I will create a new PR. Small mistake, I did rebase.

File: libp2p/host/basic_host.py [hunk 1]
```diff
@@ -288,6 +288,9 @@ async def _swarm_stream_handler(self, net_stream: INetStream) -> None:
             protocol, handler = await self.multiselect.negotiate(
                 MultiselectCommunicator(net_stream), self.negotiate_timeout
             )
+            if protocol is None:
+                await net_stream.reset()
+                raise StreamFailure("No protocol selected")
```
Comment (acul71) [inline libp2p/host/basic_host.py:293]: For error messages, can you be more verbose, something like:
```raise StreamFailure("Failed to negotiate protocol: no protocol selected")```

File: libp2p/security/security_multistream.py [hunk 2]
```diff
@@ -114,5 +117,7 @@ async def select_transport(
         else:
             # Select protocol if non-initiator
             protocol, _ = await self.multiselect.negotiate(communicator)
+            if protocol is None:
+                raise SecurityUpgradeFailure("No protocol selected")
```
Comment (acul71) [inline libp2p/security/security_multistream.py:121]: Same. 
``` raise SecurityUpgradeFailure("failed to negotiate the secure protocol")```

File: libp2p/stream_muxer/muxer_multistream.py [hunk 3]
```diff
@@ -73,14 +76,16 @@ async def select_transport(self, conn: IRawConnection) -> TMuxerClass:
...
+        protocol: TProtocol | None
         communicator = MultiselectCommunicator(conn)
         if conn.is_initiator:
             protocol = await self.multiselect_client.select_one_of(
                 tuple(self.transports.keys()), communicator
             )
         else:
             protocol, _ = await self.multiselect.negotiate(communicator)
+            if protocol is None:
+                raise MuxerUpgradeFailure("No protocol selected")
```
Comment (acul71) [inline libp2p/stream_muxer/muxer_multistream.py:88]: Same, somethink like:
```aise MuxerUpgradeFailure("failed to negotiate the multiplexer protocol")```

**Review by acul71 — COMMENTED** [review]
Thanks, @unniznd  This fix addresses a legitimate type consistency issue and improves error handling throughout the codebase. The implementation is clean, focused, and maintains backward compatibility.

Can you please **chore the error messages** text and can you **add some test** to see that the None case is well handled ?
Verify 
```python
     # basic_host.py
     raise StreamFailure("failed to negotiate protocol")
     
     # security_multistream.py  
     raise SecurityUpgradeFailure("failed to negotiate the secure protocol")
     
     # muxer_multistream.py
     raise MuxerUpgradeFailure("failed to negotiate the multiplexer protocol")
     ```
